### ISSUE TITLE
Syncing problem

### ISSUE THREAD

**Issue by arimotearipo** [langerhans/dogecoin-wallet-new#67]
I stashed away some dogecoins on this very app 3.5 years ago. Today I'm trying to access to those coins but it says that I'm still syncing. It takes forever. I tried resetting the blockchain but after waiting a full night, only ONE block has been replayed.

**Comment by zzudwa** [comment]
I was about to create same trouble ticket but you already did.
It's nothing about 3 years. 6 weeks ago it stopped synchronizing.
It connected to peers but not even trying to start download.
Seems that something updated on the peers side so this client app can't download blocks anymore.
I've tried on different devices but result is same. No matter how many months yet to synchronize 24 or 4 or 1 it's simply not moving.

I hope developer can solve the problem as really a lot of money stuck in there.

**Comment by Jademalo** [comment]
Mine is stuck too.

Tried extracting the private keys to sweep into an online wallet, but I've got absolutely no idea how to do that. The only resources I can find are links to here - https://github.com/bitcoin-wallet/bitcoin-wallet/blob/master/wallet/README.recover.md, but this is for bitcoin and I've got no idea how to use it for doge.

**Comment by zzudwa** [comment]
Ive just realized.
Peers with port :70001 doesn't work, but peer with port :70015 seems to be working. If you are lucky to find one.
![IMG_20210129_193235_042](https://user-images.githubusercontent.com/66626977/106270314-c2d82c80-6268-11eb-97f3-ff4a94008614.jpg)

**Comment by langerhans** [comment]
This is all realted to the peering issues we're currently seeing on the network. We have brought up many more nodes to help with that but DOGE is immensely popular right now so nodes have a lot more new user influx to deal with. That's also why the new wallet update isn't out yet even tho it's almost ready. All you can currently do is keeping to try syncing. Sorry about that.

**Comment by Jademalo** [comment]
Alright, thanks for the clarification. 
Are there any specific trusted addresses we can try to hopefully improve things? I'm not even finding peers, let alone recent ones.

**Comment by langerhans** [comment]
Not really, peers always have a limited amount of connections they can accept. So it's no use to send hundreds of people onto a single node.

**Comment by zzudwa** [comment]
> This is all realted to the peering issues we're currently seeing on the network. We have brought up many more nodes to help with that but DOGE is immensely popular right now so nodes have a lot more new user influx to deal with. That's also why the new wallet update isn't out yet even tho it's almost ready. All you can currently do is keeping to try syncing. Sorry about that.

Is there a chance that in new version of wallet app there will be a human friendly way to export private key? 🥺

**Comment by zendeskboss** [comment]
> Not really, peers always have a limited amount of connections they can accept. So it's no use to send hundreds of people onto a single node.

Is it possible to set up a node ourselves and connect to it? Apologies if that's a super layman question!
(only asking because I have 84 months worth to sync...ouch!)

**Comment by zzudwa** [comment]
> Not really, peers always have a limited amount of connections they can accept. So it's no use to send hundreds of people onto a single node.

Let's say I have a full node on my own private server. How can I download blockchain from there to this android wallet?

**Comment by LevitatingBusinessMan** [comment]
@technonorm wow that would be interesting, downloading the bootstrap blockchain and setting up some rogue node to serve it.

**Comment by Tschda2356** [comment]
> > This is all realted to the peering issues we're currently seeing on the network. We have brought up many more nodes to help with that but DOGE is immensely popular right now so nodes have a lot more new user influx to deal with. That's also why the new wallet update isn't out yet even tho it's almost ready. All you can currently do is keeping to try syncing. Sorry about that.
> 
> Is there a chance that in new version of wallet app there will be a human friendly way to export private key? pleading_face

This would be an incredibly useful feature. Me and some others have tried to get the private key through the seed phrase that is exported with the wallet with no luck yet. Though I am somewhat experienced with blockchain technologies, I am not as versed as I'd liked to be and I have not been able to extract the key I was looking for with tools like this one: https://iancoleman.io/bip39/ - It would be great if @langerhans would enable us to extract the key either directly through the app or leave a guide on how to do it somehow through the exported wallet file.

**Comment by LevitatingBusinessMan** [comment]
@WindowsGott50 So it should be cryptographically possible to get the privkeys with the seed phrase?

**Comment by Jademalo** [comment]
For some reason, when I can finally connect to a 1.14.2 protocol 70015 seed, it rapidly connects and disconnects from them without downloading anything.

**Comment by LevitatingBusinessMan** [comment]
@Jademalo The nodes have a set limit of concurrent connections they can have. If you leave your phone open for a while it should at some point find some peers when room clears up.

**Comment by Jademalo** [comment]
@LevitatingBusinessMan I've had it open for around 9 hours now, and not a single block of progress has occurred. I've had plenty of peers in this time, but none of them have progressed the chain on my end.

**Comment by Tschda2356** [comment]
> @WindowsGott50 So it should be cryptographically possible to get the privkeys with the seed phrase?

Yes. It seems that the seed phrase enables a pretty much infinite amount of private keys and public keys to be generated. The generation involves a kind of "path" (I am again referring to [this tool](https://iancoleman.io/bip39/)). Extended private keys and extended public keys are generated through the combination of Derivation algorithm + seed phrase + path. I guess if we have the information on what algorithm is used and which "path" it takes it is possible to find the corresponding private key to your wallet.

**Comment by LevitatingBusinessMan** [comment]
@Jademalo I know, I've had the same. Only just now I have gotten a few blocks after being stuck at the same block for over 12 hours.

**Comment by firlus** [comment]
I found a method in the code called writeKeys which basically seems to do what we want...
Maybe we can use this in a small script outside of the app to get the keys

**Comment by LevitatingBusinessMan** [comment]
@WindowsGott50 I think the seed phrase in the archive might sadly only be used for the addresses that are also in the archive, not the receive addresses.

**Comment by jga01** [comment]
> I found a method in the code called writeKeys which basically seems to do what we want...
> Maybe we can use this in a small script outside of the app to get the keys

Please.

**Comment by langerhans** [comment]
One can decrypt a backup and extract the keys, yes. There is a guide somewhere that I can't find right now. 
Excuse the lack of quick responses, I do have a day job to work too.

As for the peering problem: If you have a **fully synced** Dogecoin Core running somewhere then yes you can connect the app directly to that using the Trusted Peer feature. It would actually help us if people who already have **fully synced** clients run them with `listen=1` and open Port 22556 on their firewall.

**Comment by LevitatingBusinessMan** [comment]
@langerhans The archive only contains like 4 pubkeys with privkeys. And then some hundred public keys with no private keys. There is a seed phrase in the archive. Do you know how to recover the privkeys via the seed phrase?

I am using this fork I have to extract the archive: https://github.com/LevitatingBusinessMan/wallet-decrypt
Occasionally modifying it to my needs.

**Comment by LevitatingBusinessMan** [comment]
@langerhans and could you elaborate on how to run the dogecoin core with cli arguments like that? I can't seen to get it woring.
 I am almost on a full sync so I'll torrent the blockchain when I get it close to complete.

 Edit: default config allows listening for connections, just added my local address to the app and I am downloading blocks! @langerhans thanks a lot.

**Comment by firlus** [comment]
https://github.com/langerhans/dogecoin-wallet-new/issues/68#issuecomment-769800820

This awesome guy figured it out!!!

**Comment by langerhans** [comment]
@LevitatingBusinessMan You can just go to Settings->Network->Allow incoming connections. For a daemon client you need a dogecoin.conf file in your data director (where wallet.dat is) with the parameter I mentioned.

**Comment by zzudwa** [comment]
> @langerhans and could you elaborate on how to run the dogecoin core with cli arguments like that? I can't seen to get it woring.
> I am almost on a full sync so I'll torrent the blockchain when I get it close to complete.
> 
> Edit: default config allows listening for connections, just added my local address to the app and I am downloading blocks! @langerhans thanks a lot.

How did you added local address to the app?

**Comment by technonorm** [comment]
> > @langerhans and could you elaborate on how to run the dogecoin core with cli arguments like that? I can't seen to get it woring.
> > I am almost on a full sync so I'll torrent the blockchain when I get it close to complete.
> > Edit: default config allows listening for connections, just added my local address to the app and I am downloading blocks! @langerhans thanks a lot.
> 
> How did you added local address to the app?

Could you stick up the torrent please? I'm stuck at 12 days behind :(

**Comment by zendeskboss** [comment]
> @langerhans and could you elaborate on how to run the dogecoin core with cli arguments like that? I can't seen to get it woring.
> I am almost on a full sync so I'll torrent the blockchain when I get it close to complete.
> 
> Edit: default config allows listening for connections, just added my local address to the app and I am downloading blocks! @langerhans thanks a lot.

Could you post the torrent please? I'm stuck on 12 days behind :(

**Comment by LevitatingBusinessMan** [comment]
@zzudwa in the settings, set the "trusted peer"
@technonorm from what I know nobody is torrenting the full chain. For updates on that be on the discord. I am at 13 jan myself.

**Comment by zendeskboss** [comment]
> @zzudwa in the settings, set the "trusted peer"
> @technonorm from what I know nobody is torrenting the full chain. For updates on that be on the discord. I am at 13 jan myself.

omg I got it working! Thanks

For those looking to connect to your core client, 

- you need to port forward 22556 in your router. Find out the Internal IP address for the computer hosting core
- Accept incoming connections in your dogecoin core settings
- On your mobile wallet: settings>settings>Trusted peer- 
- Enter your internal IP address in trusted peer

You should see your connection go in. Mine's flying through now!

**Comment by LevitatingBusinessMan** [comment]
@technonorm If you aren't leaving your home, you can just use the local address of your desktop.

**Comment by zendeskboss** [comment]
> @technonorm If you aren't leaving your home, you can just use the local address of your desktop.

Thanks. I initially added the port and it failed. Didnt think to try internal without the port. Edited my instructions!

Only thing left is to sync up those last 12 days now. So close and yet so far!

**Comment by SikoSoft** [comment]
As others have stated, my app gets stuck and never progresses. My phone is 18 months behind, left on the entire night, nothing has happened yet. No progress, no new blocks.

Tried installing on my tablet. It is 21 months behind and so far also remains stuck.

Also I had tried sweeping a paper wallet... Don't do that. If you get stuck in limbo like this you may lose the money. After getting a paper wallet redeemed from dogechain.info in literally less than 30 minutes, I decided enough of this crap app, so I went to get my keys from it and use dogechain.info instead. The problem? Besides the hassle of hacking around with python tools and wallet generators even to obtain your private keys, the wallet backup doesn't even contain the address it has swept (and confirmed) to. Nice. Real nice stuff here.

**Comment by zzudwa** [comment]
> As others have stated, my app gets stuck and never progresses. My phone is 18 months behind, left on the entire night, nothing has happened yet. No progress, no new blocks.
> 
> Tried installing on my tablet. It is 21 months behind and so far also remains stuck.
> 
> Also I had tried sweeping a paper wallet... Don't do that. If you get stuck in limbo like this you may lose the money. After getting a paper wallet redeemed from dogechain.info in literally less than 30 minutes, I decided enough of this crap app, so I went to get my keys from it and use dogechain.info instead. The problem? Besides the hassle of hacking around with python tools and wallet generators even to obtain your private keys, the wallet backup doesn't even contain the address it has swept (and confirmed) to. Nice. Real nice stuff here.

Simple solution described in comments on thread https://github.com/langerhans/dogecoin-wallet-new/issues/68
The hassle with sync happened because of pumpers attack https://t.me/EarnWithYobit/101

**Comment by PardonMyWhiteness** [comment]
So... similar issue. 

The app does work.

Will find out if what sent that pizza money had left in reaches destination. Charges are 1 doge to send.

Solution: in settings under network in the app, leave that open, I take it for you it will be empty.
Then try go to your network router for your wifi etc, disconnect or reset that. Let it come back by itself and monitor back in the app if any improvement.

My guess are that isp or router eventually clogs or gets blocked, reseting router and isp connections helps refresh maybe to see it as a ordinary process.


Times been tough so if willing to donate some doge, Holla I guess.
